en/lessons/basics/basics.md

```diff
@@ -9,269 +9,197 @@ Getting started, basic data types, and basic operations.
 
 ## Getting Started
 
-### Installing Elixir
+### Installing Haskell
 
-Installation instructions for each OS can be found on elixir-lang.org in the [Installing Elixir](http://elixir-lang.org/install.html) guide.
+The recommended installation instructions for each OS can be found at
+[ghcup](https://www.haskell.org/ghcup/).
 
-After Elixir is installed, you can easily find the installed version.
+You can check the version of the Haskell compiler installed by running
+`ghc --version`:
 
-    % elixir -v
-    Erlang/OTP {{ site.erlang.OTP }} [erts-{{ site.erlang.erts }}] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]
-
-    Elixir {{ site.elixir.version }}
+    % ghc --version
+    The Glorious Glasgow Haskell Compilation System, version {{ site.ghc.version }}
 
 ### Trying Interactive Mode
 
-Elixir comes with IEx, an interactive shell, which allows us to evaluate Elixir expressions as we go.
+Your installation should come with `ghci`, an interactive shell that allows you
+to try out code in real time.
 
-To get started, let's run `iex`:
+Follow this tutorial by running `ghci`:
 
-    Erlang/OTP {{ site.erlang.OTP }} [erts-{{ site.erlang.erts }}] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]
+    GHCi, version {{ site.ghc.version }}: https://www.haskell.org/ghc/  :? for help
+    ghci>
 
-    Interactive Elixir ({{ site.elixir.version }}) - press Ctrl+C to exit (type h() ENTER for help)
-    iex>
+To use `Text` in Haskell, type in:
 
-Note: On Windows PowerShell, you need to type `iex.bat`.
+```haskell
+ghci> :set -XOverloadedStrings
+ghci> import qualified Data.Text as T
+```
```

Comment:
I don't buy the `Text` type for an introductory material on today's Haskell: it adds a heck lot of fuss like all those type annotations -- they will look very confusing to a novice. Someday, maybe when something like [this proposal](https://github.com/ghc-proposals/ghc-proposals/pull/409) happens, I could change my mind.

Reply:
I'm not sure I want to introduce a datatype to beginners that I wouldn't recommend using. However, I can remove a lot of the type annotations by using functions such as `T.length` and `T.putStrLn`. 

@Kleidukos: what should I do here?

Reply:
I don't understand this anxiety about `String`: if you don't write a text processing application (text as in books, novels, etc.), and, say, simply need to print a greeting to a user, why wouldn't you use `String`? Everything should be judged by the cost/benefit ratio: if your program uses just several short strings, there's no need to pull in `Text`. Note that `Char` (and therefore `String`) in Haskell is [Unicode-aware](https://www.haskell.org/onlinereport/haskell2010/haskellch6.html#x13-1190006.1.2). So we're only really trading performance here. Is this tutorial the right place to optimize for performance, or maybe we'd want to prioritize usability?

There would be a place to discuss all intricacies of all string types, I hope: `text` is just a part of the story, and somewhere we'd need to introduce ByteStrings...

Reply:
> if you don't write a text processing application (text as in books, novels, etc.), and, say, simply need to print a greeting to a user, why wouldn't you use String?

As a general-purpose textual data container, String is very bad. String has a very poor API, performances that go against what people know in other languages except C and the old Erlang strings, and I am afraid that if their utility disappears once we go beyond printing a greeting to a user, they shouldn't be mentioned at all.

I think linked lists should be taught as its own subject, and it should be mentioned that a common idiom is to have ad-hoc textual data represented as a list of `Char`, and outlining that it may be a nice data structure to implement templates that must be changed frequently[¹](https://samuelmullen.com/articles/phoenix-templates/).

> if your program uses just several short strings, there's no need to pull in Text.

I frequently uses several short strings and `Text` is an absolute must-have because I have _no wish_ to replicate its API on `String`. This is a matter of ergonomics and developer experience, not just data locality in contiguous slots of memory.

Reply:
> However, I can remove a lot of the type annotations by using functions such as T.length and T.putStrLn.

@TrueBoxGuy Let us do so then